src/extension/TeamCityEventListener.cs

```diff
@@ -37,7 +37,7 @@ namespace NUnit.Engine.Listeners
     // purely documentary since engines prior to 3.4 do not
     // check the EngineVersion property and will try to
     // load this extension anyway.
-    [Extension(Enabled = false, EngineVersion = "3.4")]
+    [Extension(Enabled = false, EngineVersion = "3.5")]
```

Comment:
I agree with @CharliePoole that our intention was that the `EngineVersion` specify the lowest version of the engine that the extension will work with, not the prefered version. If users have `NUnit.ConsoleRunner` 3.4.1 installed, then see that you have released a newer version of this extension, they will update and it will stop working.

You say they will install using a ZIP containing everything. Most users don't do this, most users install using NuGet.

I would hope that the changes you made to the V2 framework driver and to the engine are not co-dependent on fixes here. I assume that those are additional fixes that you would like to see users using, but they will not stop this extension from working?